x-pack/plugins/maps/public/shared/layers/layer.js

```diff
@@ -101,8 +104,66 @@ export class AbstractLayer {
     return this._descriptor.label ? this._descriptor.label : '';
   }
 
-  getIcon() {
-    console.warn('Icon not available for this layer type');
+  getCustomIconAndTooltipContent() {
+    return {
+      icon: (
+        <EuiIcon
+          size="m"
+          type={this.getLayerTypeIconName()}
+        />
+      )
+    };
+  }
+
+  getIconAndTooltipContent(currentZoomLevel) {
+    let icon;
+    let tooltipContent;
+    if (this.hasErrors()) {
+      icon = (
+        <EuiIcon
+          aria-label={i18n.translate('xpack.maps.layer.loadWarningAriaLabel', { defaultMessage: 'Load warning' })}
+          size="m"
+          type="alert"
+          color="warning"
+        />
+      );
+      tooltipContent = this.getErrors();
+    } else if (this.isLayerLoading()) {
+      icon = (<EuiLoadingSpinner size="m"/>);
```

Comment:
initialize `tooltipContent` with `null`. It always feels a little weird passing around undefined values intentionally